### PR TITLE
Changes in the code that uses BayesianTools package

### PR DESCRIPTION
**Pull request by istfer**
I did some minor changes here and there to accommodate the latest updates in the package

@araiho is up for review

### CODE REVIEW

File: modules/assim.batch/R/pda.bayesian.tools.R [hunk 1]
```diff
@@ -125,49 +116,37 @@ pda.bayesian.tools <- function(settings, params.id=NULL, param.names=NULL, prior
...
+
   ## Apply BayesianTools specific settings
   bt.settings=pda.settings.bt(settings)
   
   ## central function in BayesianTools
-  out <- runMCMC(bayesianSetup = bayesianSetup, sampler = settings$assim.batch$bt.settings$sampler, settings = bt.settings)
+  out <- runMCMC(bayesianSetup = bayesianSetup, sampler = sampler, settings = bt.settings)
+  
+  # save the out object for further inspection
+  save(out, file=file.path(settings$outdir, paste0(sampler, '_out.Rdata')))
```
Comment (araiho) [inline modules/assim.batch/R/pda.bayesian.tools.R]: @istfer Does any other script use this output file?

File: modules/assim.batch/R/pda.utils.R [hunk 2]
```diff
@@ -740,36 +740,25 @@ pda.settings.bt <- function(settings){
...
   
   iterations = as.numeric(settings$assim.batch$bt.settings$iter)
-  optimize = settings$assim.batch$bt.settings$optimize
-  if(!is.null(settings$assim.batch$bt.settings$consoleUpdates)) consoleUpdates = as.numeric(settings$assim.batch$bt.settings$consoleUpdates) else consoleUpdates = NULL
-  parallel = settings$assim.batch$bt.settings$parallel
-  adapt = settings$assim.batch$bt.settings$adapt
-  if(!is.null(settings$assim.batch$bt.settings$adaptationInverval)) adaptationInverval = as.numeric(settings$assim.batch$bt.settings$adaptationInverval) else adaptationInverval=NULL
-  if(!is.null(settings$assim.batch$bt.settings$adaptationNotBefore)) adaptationNotBefore = as.numeric(settings$assim.batch$bt.settings$adaptationNotBefore) else adaptationNotBefore=NULL
-  initialParticles=list("prior",as.numeric(settings$assim.batch$bt.settings$n.initialParticles))
+  if(!is.null(settings$assim.batch$bt.settings$optimize)) optimize = settings$assim.batch$bt.settings$optimize else optimize=TRUE
```
Comment (araiho) [inline modules/assim.batch/R/pda.utils.R]: This is good. I should add something like this to sda eventually.

File: modules/assim.batch/R/pda.utils.R [hunk 3]
```diff
@@ -740,36 +740,25 @@ pda.settings.bt <- function(settings){
...
   if(sampler == "Metropolis") {
-    bt.settings <- list(iterations = iterations, adapt = adapt, DRlevels = DRlevels, gibbsProbabilities = gibbsProbabilities, 
-                     temperingFunction = temperingFunction, optimize = optimize)
+    bt.settings <- list(iterations = iterations, optimize = optimize, DRlevels = DRlevels, adapt = adapt, 
+                        adaptationInverval=adaptationInverval, adaptationNotBefore=adaptationNotBefore,
+                        gibbsProbabilities = gibbsProbabilities, consoleUpdates=consoleUpdates)
   } else if(sampler %in% c("AM", "M", "DRAM", "DR")) {
     bt.settings = list(iterations = iterations, startValue = "prior")
-  } else if(sampler %in% c("DE", "DEzs")) {
+  } else if(sampler %in% c("DE", "DEzs","DREAM", "DREAMzs")) {
```
Comment (araiho) [inline modules/assim.batch/R/pda.utils.R]: What is DREAM?

**Comment by araiho** [comment]
@mdietze and @istfer Looks good to me! I just asked a few questions for my own understanding.

File: modules/assim.batch/R/pda.utils.R (hunk 3 above)
Comment (mdietze) [inline modules/assim.batch/R/pda.utils.R]: DiffeRential Evolution Adaptive Metropolis

File: modules/assim.batch/R/pda.bayesian.tools.R [hunk 4]
```diff
@@ -125,49 +116,37 @@ pda.bayesian.tools <- function(settings, params.id=NULL, param.names=NULL, prior
...
   
-  # SHOULD WE USE PACKAGE'S PLOT METHODS?
-  n.row = as.numeric(settings$assim.batch$bt.settings$iter)
+  samples=getSample(out, parametersOnly = T) # getSample{BayesianTools}
+  n.row=length(samples)/n.param
+  params <- matrix(rep(parm,n.row), nrow=n.row, ncol=n.param.all, byrow=T)
+  params[, prior.ind] <- samples
   
-  if(sampler=="SMC"){
-    params <- matrix(rep(parm,bt.settings$initialParticles[[2]]), nrow=bt.settings$initialParticles[[2]], ncol=n.param.all, byrow=T)
```
Comment (mdietze) [inline modules/assim.batch/R/pda.bayesian.tools.R]: Was there a reason to drop SMC as an option?

File: modules/assim.batch/R/pda.utils.R [hunk 5]
```diff
@@ -740,36 +740,25 @@ pda.settings.bt <- function(settings){
...
-  if(!is.null(settings$assim.batch$bt.settings$adaptationNotBefore)) adaptationNotBefore = as.numeric(settings$assim.batch$bt.settings$adaptationNotBefore) else adaptationNotBefore=NULL
-  initialParticles=list("prior",as.numeric(settings$assim.batch$bt.settings$n.initialParticles))
+  if(!is.null(settings$assim.batch$bt.settings$optimize)) optimize = settings$assim.batch$bt.settings$optimize else optimize=TRUE
+  if(!is.null(settings$assim.batch$bt.settings$consoleUpdates)) consoleUpdates = as.numeric(settings$assim.batch$bt.settings$consoleUpdates) else consoleUpdates = max(round(iterations/10),100)
+  if(!is.null(settings$assim.batch$bt.settings$adapt)) adapt = settings$assim.batch$bt.settings$adapt else adapt=TRUE
+  if(!is.null(settings$assim.batch$bt.settings$adaptationInverval)) adaptationInverval = as.numeric(settings$assim.batch$bt.settings$adaptationInverval) else adaptationInverval=max(round(iterations/100*5),100)
+  if(!is.null(settings$assim.batch$bt.settings$adaptationNotBefore)) adaptationNotBefore = as.numeric(settings$assim.batch$bt.settings$adaptationNotBefore) else adaptationNotBefore=adaptationInverval
```
Comment (bpbond) [inline modules/assim.batch/R/pda.utils.R]: Note there's a wide variety of coding styles in these commits (e.g. `<-` versus less-desirable `=`, `if`s with and without braces/newlines, use of camelCase versus dot.case for variable names) that detracts from readability. Might consider a Pecan style guide at some point.

Comment (mdietze) [inline modules/assim.batch/R/pda.utils.R]: We do have a style guide! https://pecan.gitbooks.io/pecan-documentation/content/developers_guide/Coding_style.html

But clearly we're not enforcing it. Thanks for the reminder.

Comment (bpbond) [inline modules/assim.batch/R/pda.utils.R]: Always happy to make work for others. :)

Comment (dlebauer) [inline modules/assim.batch/R/pda.utils.R]: @istfer I'd suggest the function `ifelse`. 

@mdietze perhaps we can add a link to the CONTRIBUTING and PULL_REQUEST files. And start bringing it up in code reviews. The basics are pretty simple.

File: modules/assim.batch/R/pda.bayesian.tools.R (hunk 1 above)
Comment (istfer) [inline modules/assim.batch/R/pda.bayesian.tools.R]: @araiho no, at least not yet, but it stores helpful information (the settings that produced that output, summaries such as acceptance rate, likelihoods and priors). I find it very handy for my own work to save this out object. Maybe I can also save everything in such a list for our natively implemented algorithms

File: modules/assim.batch/R/pda.bayesian.tools.R (hunk 4 above)
Comment (istfer) [inline modules/assim.batch/R/pda.bayesian.tools.R]: I realized that their getSample function handles different output formats (Metropolis has iteration number +1 rows, DREAM has 3 chains, SMC has particles etc) of each algorithm so I don't have to specify different cases. this version takes care of SMC outputs too

File: modules/assim.batch/R/pda.utils.R (hunk 5 above)
Comment (istfer) [inline modules/assim.batch/R/pda.utils.R]: good point. camelCase vs dot.case is indeed due to different coding styles, the former is the style they use in the package, the latter is what we use in PEcAn. The rest is my bad. I should add the the shortcut for `<-` on my keyboard at the desktop too :)

@dlebauer that I'll do! this bit started as only `if`s then I added the `else`s... I'll change it

**Pull request merged**